lib/pipeline/resources/auth.rb

```diff
@@ -0,0 +1,33 @@
+module Pipeline
+  class Auth < Pipeline::Resource
+    include ActiveResource::Singleton
+
+    def self.authenticate(email, password, mfa_code = nil, options = {})
+      add_keys(options)
+      options.merge!(email_or_username: email, password: password, mfa_code: mfa_code)
+
+      self.include_root_in_json = false
+      auth = create(options)
+      self.include_root_in_json = true
+
+      if auth.respond_to?(:token)
+        configure_jwt_token(auth)
+        Pipeline::User.find(auth.user.id)
+      else
+        configure_api_key(auth)
+        # For some reason, this path does not return the user within a hash, so the user id is at the top level.
+        Pipeline::User.find(auth.id)
```

Comment:
Do we need to fetch the user here?

Should we simply return the auth hash...
```
auth.respond_to?(:token) ? auth.user : auth
```

And let the integrator decide if fetching the actual user is required?

Very possible I'm misunderstanding the intent also 😄

Reply:
The object returned was either not actually a user (in the case of jwt because there was no root) or in the case of the api_key, it was a user object, but ActiveRecourse didn't realize it existed. A fetch was the cleanest way to get ActiveResource to treat it correctly.

Reply:
I would prefer to use `Pipeline::User.authenticate(email:, password:, mfa_code:)`. But it has to post to `/api/v3/auth`.

Adding the `authenticate` method in the `Pipeline::User` model, and adding `post(:auth)` inside of `User` would try to call `/api/v3/users/auth`. If we could get this to post to the right endpoint, then I'll gladly get rid of `Pipeline::Auth` model, which is really not a valid model.